### PR TITLE
Fix loading OnetoOne self referencing entity

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java [hunk 1]
```diff
@@ -998,6 +998,22 @@ public EntityFetch generateFetch(
...
+					fetchParent,
+					this,
+					fetchablePath,
+					foreignKeyDescriptor.createDomainResult(
+							fetchablePath,
+							parentTableGroup,
+							ForeignKeyDescriptor.Nature.TARGET,
+							creationState
+					),
+					true,
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:1012]: Is this really always a select by unique key?

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java [hunk 2]
```diff
@@ -998,6 +998,22 @@ public EntityFetch generateFetch(
...
 
+		if ( cardinality == Cardinality.ONE_TO_ONE && bidirectionalAttributeName != null && fetchParent.getReferencedMappingType() == getEntityMappingType() ) {
+			return new EntityFetchSelectImpl(
+					fetchParent,
+					this,
+					fetchablePath,
+					foreignKeyDescriptor.createDomainResult(
+							fetchablePath,
+							parentTableGroup,
+							ForeignKeyDescriptor.Nature.TARGET,
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:1009]: It seems to me, that this is not entirely correct. See how the side nature is determined below.

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java [hunk 3]
```diff
@@ -998,6 +998,22 @@ public EntityFetch generateFetch(
...
+							creationState
+					),
+					true,
+					creationState
+			);
+		}
+
 		if ( fetchTiming == FetchTiming.IMMEDIATE && selected ) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:1017]: Maybe it's better to exclude the above case to entering this if-branch instead as the below code block should handle delayed/select fetching?

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java (hunk 2 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:1009]: if I use the same way, the test fails may be the issue is the wrong value of the `this.sideNature`

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:1012]: it seems so for the case case of ` cardinality == Cardinality.ONE_TO_ONE && bidirectionalAttributeName != null && fetchParent.getReferencedMappingType() == getEntityMappingType()`

**Review by beikov — APPROVED** [review]

**Pull request merged**